packages/settings/src/settings.js

```diff
@@ -222,4 +223,16 @@ export default {
      * @default false
      */
     ROUND_PIXELS: false,
+
+    /**
+     * PixiJS generates some efficient webgl-related functions, but not all platforms support it.
+     * Default value is auto-detected.
+     *
+     * @static
+     * @constant
+     * @memberof PIXI.settings
+     * @type {boolean}
+     * @default true
+     */
+    CAN_GENERATE_NEW_FUNCTION: canGenerateNewFunction(),
```

Comment:
I don’t quite see the use case for another setting here. Just use the feature detection. The use case where someone needs to set this explicitly seem so remote. And in that case they could override canGenerateNewFunction.

Reply:
> And in that case they could override canGenerateNewFunction.

We already have mobile detection and other stuff detection.

Usecase - debug of the generator. Also I have several ideas for other generators, so better use it in one place.

Reply:
I can put it in `utils` :)

Reply:
I disagree with your rationale here. I feel like you often add more than is necessary. Just solve the original problem as succinctly as possible. When you introduce new generators in the future, then we can discuss on adding a debug flag then. Just cache this as a private variable. No new unnecessary settings or APIs for v5.